index.bs

```diff
@@ -390,6 +399,10 @@ type: dfn
     following algorithm:
 
     <ol>
+      <li>
+        Return <strong>blocked</strong> if <var>request</var>
+        <a for="request">URL</a>'s <a for="url">host</a> is a <a>non-loopback IP address</a>.
+      </li>
```

Comment:
Looking at this again it seems this doesn't allow for overrides as per the next bullet point. It would also fail targeted navigations of popups to a public IP address. Wouldn't it be better to leave this out and rely on "is a potentially trustworthy URL" in the next list item?

Reply:
That's an interesting point. I think the initial issue @carlosjoan91 mentioned about with this was that we didn't want to introduce a loophole in mixed-content, for the case where we would not auto-upgrade the request, but not block it either. Indeed, for the "upgradable content", I guess we do not want to regress what we currently have. For instance, what will currently happen with today's spec is that we will auto-upgrade the request, and we most likely won't be able to get a certificate for the IP address, so the request will eventually fail, which is precisely what we want to avoid here.

If you completely remove what's currently in `4.4`, we will most likely end-up with the fetch succeeding for requests like `http://8.8.8.8/music.mp3` (audio type), which we want to avoid since it was previously an auto-upgraded requests.

I think what we can then do is add a note in `4.4` stating that the user-agent might want to block upgradable content if the URL's host is an non-loopback IP address? It seems that the choice of the behaviour is let to the user agent concerning upgradable content anyways, so maybe doing just this is enough...

What do you think @annevk @carlosjoan91 ?

Reply:
That was my initial thoughts as well. But @carlosjoan91 had concerns about this as it might bring-up a loophole for upgradeable content.

The thing is, this section is about `blockable content`, but yet by not upgrading IP addresses in section `4.1`, we might end-up with a regression as opposed to before, where we would most likely fail when trying to load an https request with an IP address (because lacking a certificate). Ideally, we would remove that part here, since it only concerns blockable content, but add some words somewhere about blocking fetches for upgradeable content when the URL's host in a non-loopback IP address. What do you think @annevk @carlosjoan91 ?

Reply:
Pointer for the loophole?

Reply:
IIRC. Currently, for upgradeable mixed-content, we auto-upgrade the request which URL's host is an IP address, so we won't even bother with the block fetch, since this would be upgrade. However, this would most likely fail because we won't have a TLS certificate for the IP address.

With this change, we would not auto-upgrade the request, and thus will probably load the upgradeable content as-is, since this is content we avoid blocking because it could "break the internet" basically. I just want to make sure in this particular case we would still block the fetch.

Reply:
@annevk I reworded the sentence, what do you think?

Reply:
Upgradeable (and blockable) are not terms that work in normative algorithms as they require a lot of implicit context to be understood.

I don't really understand though.

We keep the change that IP addresses don't get their scheme changed.

Then when it comes time to block, we continue to allow loopback IP addresses by virtue of potentially trustworthy URLs and we also continue to not block any IP address in case of popup navigations, end user settings, or the special settings object case. All other cases would still get blocked as that's the very next step.

What am I missing?

Reply:
Ok, sorry, I confused myself around upgradeable and blockable content. It's true that they do not appear in the algorithm anyways, so there is no need to mention that.

I also feel like what to do in the upgradeable part is kind of left to the user-agent anyways, so I guess whether or not literal IP addresses that didn't get auto-upgraded should be blocked is also left to it.

I think we both agree on the algorithm part anyways, so I changed accordingly, thanks!